integr8ly/integreatly-operator
# RHOAM mutlitenant - set IDP name via envar

# Issue link
https://issues.redhat.com/browse/MGDAPI-2538

# What
Let the IDP name be set via the envar so that when a tenant logs in via given IDP, his routes and integration with SSO is created

# Verification steps
Install RHOAM Multitenant from this branch using:
- `INSTALLATION_TYPE=multitenant-managed-api make cluster/prepare/local`
- `INSTALLATION_TYPE=multitenant-managed-api make code/run`
At this stage the testing IDP installation is looking for is rhd.
Once RHOAM installs, create testing IDP by:
- edit the email from ./scripts/templates https://github.com/integr8ly/integreatly-operator/blob/master/scripts/user-template.yml#L22 to end in rhmi.io instead of example.com
- run the IDP prepare script against your cluster with `"NUM_REGULAR_USER=5 NUM_ADMIN=0 PASSWORD=Password1 REALM_DISPLAY_NAME=\"rhd\" REALM=\"rhd\" ./scripts/setup-sso-idp.sh"
- log in to osd as kubeadmin and open 3scale routes 
- in another (private window) login to the cluster using rhd IDP test-user01
- After about 2-5 minutes, routes for test-user01 should appear in 3scale, try logging in to user01 admin route and confirm that logging in via IDP is working
- turn of the operator
- run the operator with `INSTALLATION_TYPE=multitenant-managed-api IDENTITY_PROVIDER_NAME=testing-idp make code/run` 
- repeat login process but this time, for test-user02 and confirm that after about 10 minutes, new routes are not available
- turn off the operator
- run the operator with `INSTALLATION_TYPE=multitenant-managed-api IDENTITY_PROVIDER_NAME=rhd make code/run`
- confirm that the routes for test-user02 become available and you are able to log in to 3scale using rhd IDP (3scale integration with IDP worked)

Possible issues:
There's a known bug that sometimes, creation of tenant can respond with 500 error leading to no access token being available. 
If when running the operator with rhd as IDP routes are not created, please reach out to me to debug further.
